lmdeploy/turbomind/deploy/target_model/base.py

```diff
@@ -136,26 +136,37 @@ def export_config(self) -> None:
 
     def export_weight(self, param: torch.Tensor, name: str) -> None:
         """export turbomind weight."""
+
+        def _tofile(tensor, path):
+            """to file."""
+            if tensor.dtype == torch.bfloat16:
+                tensor = tensor.view(torch.half)
+            tensor.contiguous().cpu().numpy().tofile(path)
+
         if self.to_file:
-            if param.dtype in [torch.float, torch.bfloat16]:
+            weight_type = self.cfg.weight_type
```

Comment:
I didn't get this change. Could you explain more in detail?

Reply:
```python
            if param.dtype in [torch.float]:
                param = param.half()
```
turbomind only accepts 2 bytes scalar type. Float tensor should be cast to half so we can export it.

```python
            if weight_type == 'fp16' and param.dtype == torch.bfloat16:
                param = param.half()
```
The default param dtype of qwen is bf16. What if the device does not support bf16, param should be cast to half.

```python
            _tofile(param, osp.join(self.out_dir, name))
```
export param to file. `.view(dtype)` would view param as a tensor of dtype, without changing the physical memory.